networking.md: add a small paragraph on exposing VLAN trunk link to VM (#453)
xcp-ng/xcp-ng-org: docs/networking/networking.md

That's it!

### Trunk link for VMs

> It is also possible to directly expose to the VM the trunk link.

Reviewer: ```suggestion
It is also possible to directly expose the trunk link to the VM.
```

The VIF should be configured to use the raw network (named by default "Pool-wide network associated with ethN").

> It will get all the 802.1Q packets on the wire, and be able to send tagged packets for any VLAN.

Reviewer 2: I assume that if the pool has a network on the same vlan as one used in the trunking, that won't work because of the bridges setup, at least on the same host, or am I missing something? If that's the case we should document it.

Author: if I properly understood what you are describing, it should work.

considering a pool with only two networks ("eth0" reserved for management, and "eth1" for VMs), you could have the following scenario:
- VM-a with "VLAN 100 over eth1"
- VM-b with "eth1" trunk, with VLANs managed by the VM
- VM-c with "VLAN 100 over eth1"

VM-a and VM-b doesn't know there is a VLAN under the wood.
if VM-b create a "VLAN 100" interface, VM-a, VM-b and VM-c will be able to communicate.

the OVS subsystem will create and configure the right ports on `xenbr1` bridge, with access port (vlan 100) for VM-a and VM-c, and one trunk port for VM-b. If VM-b sends 802.1Q packets tagged vlan 100, the packet will be properly managed by OVS, and send untagged to VM-a and VM-c. If VM-b sends 802.1Q packets tagged vlan 200, the packet will not be send to neither VM-a and VM-c.

In Xen-Orchestra, such networks have their VLAN property set to `None` (as opposed to `0` for untagged ports and `1` to `4095` for tagged ports, where  802.1Q tagged packets are dropped).

:::warning

> Be careful with the link's [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit), since VLAN needs small overhead (4 bytes), else packets could be fragged or even dropped.

Reviewer 2: shouldn't fragged be fragmented instead?

> Be careful with the link's [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit), since VLAN needs small overhead (4 bytes), else packets could be fragmented or even dropped.

Reviewer 3: ```suggestion
Be careful with the link's [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit), since VLAN tagging adds a small overhead (4 bytes). Otherwise, packets may be fragmented or even dropped.
```

> It needs proper configuration on the switch, the network in XCP-ng, and the VM's interface.

Reviewer 3: ```suggestion
It needs to be properly configured on the switch, within the XCP-ng network, and on the VM’s network interface.
```

That's it!

### Trunk link for VMs

It is also possible to directly expose the trunk link to the VM.
The VIF should be configured to use the raw network (named by default "Pool-wide network associated with ethN").
It will get all the 802.1Q packets on the wire, and be able to send tagged packets for any VLAN.

In Xen-Orchestra, such networks have their VLAN property set to `None` (as opposed to `0` for untagged ports and `1` to `4095` for tagged ports, where  802.1Q tagged packets are dropped).

:::warning

> Be careful with the link's [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit), since VLAN tagging adds a needs small overhead (4 bytes).

Reviewer 3: ```suggestion
Be careful with the link's [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit), since VLAN tagging adds a small overhead (4 bytes).
```
